app/src/lib/fuzzy-find.ts

```diff
@@ -12,36 +12,48 @@ function score(str: string, query: string, maxScore: number) {
   return fuzzAldrin.score(str, query, undefined, options) / maxScore
 }
 
+export interface IMatches {
+  readonly title: ReadonlyArray<number>
```

Comment:
I think we should re-work this interface to 

```ts
interface IMatch {
  readonly title: number
  readonly subTitle: string
}
```
That way when we pass matches, we are passing an array of objects that don't require us to use an index to relate to pieces of data together.

For example, If we want to get the title and the subtitle of a match, we could do the following:

```ts
interface IMatch {
  readonly title: number
  readonly subTitle: string
}

function foo(matches: ReadonlyArray<IMatch>) {
  const match = matches[0]

  // match contains all the data we need
  console.log(`title: ${match.title}, subtitle: ${match.subtitle}`)
}
```
As opposed to:
``` ts
interface IMatches {
  readonly title: ReadonlyArray<number>
  readonly subTitle: ReadonlyArray<string>
}

function bar(matches: ReadonlyArray<IMatches>) {
  // we have to index into two separate arrays
  const title = matches.title[0]
  const subtitle = matches.subtitle[0]

  console.log(`title: ${title}, subtitle: ${subtitle}`)
}
```

Reply:
Not sure I follow the proposed change here, as `fuzzAldrin.match` returns the `ReadonlyArray<number>` which represents the indexes of a string to highlight all the way down in `HighlightText` - are we going to change that representation to something else?

Reply:
No we are not. We talked this over a bit earlier and I'm fine with the change.

Reply:
For more context, I was under the impression that these two properties needed to be related somehow, but after going deeper into how this is used, I realized that there is no need to relate the two. @damaneice and I both agreed that naming is hard and that's what contributed to my confusion. He cleared it up for me though.